src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java

```diff
@@ -348,7 +348,7 @@ public Launcher createLauncher(TaskListener listener) {
                             ModelHyperlinkNote.encodeTo("/computer/" + getNodeName(), getNodeName()),
                             getTemplate().getDisplayName())
                     );
-                    listener.getLogger().println(getTemplate().getDescriptionForLogging());
+                    listener.getLogger().println(getTemplate().getDescriptionForLogging(this));
```

Comment:
Maybe just delete this line and have `KubernetesLauncher.launch` print the YAML of its `pod` as it is actually about to create it?

```java
listener.getLogger().printf("Created Pod: %s/%s%n", namespace, podId);
```

is where I would expect the YAML to be shown.

Reply:
The problem is that `KubernetesLauncher.launch` doesn't have access to the build logs, it goes into the agent logs and no one ever read those (and they are lost as soon as the build is over)

Reply:
Ah, forgot we were dealing with two different `TaskListener`s.